### ISSUE TITLE
More functions on "home screen" in Home Assistant

### ISSUE THREAD

**Issue by nilskt** [aglehmann/home_assistant_ecovent#2]
Are you planning to add more functions to the fan controllers? For example manually regulating the speed (from 1-100%) or changing direction of the fan? 

Thanks again for making support for these fans in Home Assistant!

**Comment by aglehmann** [comment]
Yeah, there are a few things on my to do list and manually setting the speed is one of them. That feature is already present in the python library (pyEcovent). The thing is that there is no good way of showing that feature in the lovelace UI with current "Fan" card, at least as far as I know. So i think I might have to create a custom lovelace card specifically for these units, but I haven't had the time yet. It would obviously be the best solution to show all available features in a single card.

About changing direction, you can actually do this in the current component. It contains a custom service just for that. In the developer menu under services you should see a service named "fan.ecovent_set_airflow" which can be triggered to change between the three fan modes. It is described in the readme, but I haven't gotten around to providing examples on how to use this. The component show the airflow (which I have named the fan direction) as a state attribute. This state attribute can be extracted using templates in HA and then used for automations and/or "input_select" cards. It's not optimal, but with the fan card in lovelace there isn't an option to show this feature. These fans are sort of in the middel between fan and climate component, where the latter has the option to control fan direction directly in the lovelace card. I chose a fan component to avoid any confusion about what these units do.

**Comment by nilskt** [comment]
Thanks for information regarding the fan direction. This is good news and I will try that feature! 
I am of course also  looking foreward for your future development! Especially your plans for improved UI plans (the custom card), sounds great!
Thanks again and good luck  :-)

**Comment by gavlexx** [comment]
Hi! Any news about more features in your pyEcovent - HA integration?
Now I can manipulate with 3 fan modes but cannot do anything with speed. Is there any code for this in recent version or not?

As for examples using your code in HA within automation , here it is:
`- alias: vents_test
  initial_state: 'on'
  trigger: 
  - platform: state
    entity_id: sensor.0x000000000000_action
    to: 'flip90'
  action:
    - service: fan.ecovent_set_airflow
      data_template:
        entity_id: fan.bedroom_fan
        airflow: air_supply` 

But "speed: low" doesn't work...

**Comment by aglehmann** [comment]
Hi, setting speed is already natively supported by the HA custom component. In your example you are using the custom service that ships with the custom component, setting speed is however possible to set using existing services within HA. Use the service `"fan.set_speed"` and then fill out the required values for that service:
```
"entity_id: fan.bedroom_fan"
"speed: low"
```

Since this component is somewhere between a HVAC entity and Fan entity in HA there isn't a good default card for it to control all possible settings and I have not yet gotten around to that yet.

**Comment by gavlexx** [comment]
Thanks a lot! It works :) Turn on a off also.
How can I modify your script a bit to use manual speed in automations?
Cannot understand how does it use your pyEcovent module? Nothing here found in this component to import fan object from another repository...
